File: EPZEU/.NET/EPZEU.Core/Nomenclatures/Models/NKID.cs
namespace EPZEU.Nomenclatures.Models
{
    /// <summary>
    /// Класификатор на икономически дейности.
    /// </summary>
    public class NKID
    {
        /// <summary>
        /// Идентификатор на класификатора.
        /// </summary>
        public int ID { get; set; }

        /// <summary>
        /// Код на класификатора.
        /// </summary>
        public string Code { get; set; }

        /// <summary>
        /// Текст на класификатора.
        /// </summary>
        public string Text { get; set; }

        public int? ParentID { get; set; }
    }
}

File: EPZEU_CR/.NET/EPZEU.CR.Web.App/ClientApp/EPZEU.Core/Models/NKID.ts
import { observable } from 'mobx'
import { TypeSystem, BaseDataModel } from 'Cnsys.Core'
import { moduleContext } from '../ModuleContext'

@TypeSystem.typeDecorator('NKID', moduleContext.moduleName)
export class NKID extends BaseDataModel {

    @observable private _id: number = 0;
    @observable private _code: string = null;
    @observable private _text: string = null;
    @observable private _parentID: number = 0;

    constructor(obj?: any) {
        super(obj)

        this.copyFrom(obj);
    }

    @TypeSystem.propertyDecorator('number')
    public set id(val: number) {
        this._id = val;
    }

    public get id(): number {
        return this._id;
    }

    @TypeSystem.propertyDecorator('string')
    public set code(val: string) {
        this._code = val;
    }

    public get code(): string {
        return this._code;
    }

    @TypeSystem.propertyDecorator('string')
    public set text(val: string) {
        this._text = val;
    }

    public get text(): string {
        return this._text;
    }

    @TypeSystem.propertyDecorator('number')
    public set parentID(val: number) {
        this._parentID = val;
    }

    public get parentID(): number {
        return this._parentID;
    }
}